File: TPBoardAngular/src/app/services/table.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Table } from '../interfaces/table.model';

@Injectable({
  providedIn: 'root'
})
export class TableService {
  private apiUrl = 'https://localhost:7134/api/Table';

  constructor(private http: HttpClient) { }

  getAllTables(): Observable<Table[]> {
    return this.http.get<Table[]>(`${this.apiUrl}/GetAllTables`);
  }

  getTableById(id: number): Observable<Table> {
    return this.http.get<Table>(`${this.apiUrl}/GetTableById/${id}`);
  }
  getTablesByProjectId(projectId: number): Observable<Table[]> {
    return this.http.get<Table[]>(`${this.apiUrl}/GetTablesByProjectId/${projectId}`);
  }
  createTable(table: Table): Observable<Table> {
    return this.http.post<Table>(`${this.apiUrl}/CreateTable`, table); 
  }

  updateTable(table: Table): Observable<Table> {
    return this.http.put<Table>(`${this.apiUrl}/UpdateTable/${table.id}`, table);
  }

  deleteTable(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/DeleteTable/${id}`);
  }
}

File: TPBoardWebApi/Services/TableService.cs
using TPBoardWebApi.Interfaces;
using TPBoardWebApi.Models;

namespace TPBoardWebApi.Services
{
    public class TableService : ITableService
    {
        private readonly IUnitOfWork _unitOfWork;

        public TableService(IUnitOfWork unitOfWork)
        {
            _unitOfWork = unitOfWork;
        }

        public IEnumerable<Table> GetAllTables()
        {
            return _unitOfWork.Tables.GetAll();
        }

        public Table GetTableById(int id)
        {
            return _unitOfWork.Tables.GetById(id);
        }

        public void CreateTable(Table table)
        {
            _unitOfWork.Tables.Add(table);
            _unitOfWork.Save();
        }

        public void UpdateTable(Table table)
        {
            _unitOfWork.Tables.Update(table);
            _unitOfWork.Save();
        }

        public void DeleteTable(int id)
        {
            var table = _unitOfWork.Tables.GetById(id);
            if (table != null)
            {
                _unitOfWork.Tables.Delete(table);
                _unitOfWork.Save();
            }
        }
    }
}
